test/libsolidity/semanticTests/functionTypes/selector_ternary.sol

```diff
@@ -0,0 +1,10 @@
+contract C {
+	function f() public {}
+	function g() public {}
+	function h(bool c) public returns (bytes4) {
+		return (c ? this.f : this.g).selector;
```

Comment:
This PR will (should) disallow these cases, but we apparently don't have them covered:
- `condition ? C.f : C.f`
- `condition ? C.f : C.g` (where `f` and `g` are both `external`)
- `condition ? C.f : D.f` (`external` functions from different contracts)
- `condition ? L.f : L.g`
- `condition ? L.f : K.f` (functions from different libraries)
- `condition ? I.f : J.f` (functions from different interfaces)

If we had these, we would not have missed the ICE I posted in the other comment.

And these will not be disallowed so they need semantic tests to ensure we choose the right function:
- `condition ? g : C.f` / `condition ? C.f : g` (`f` and `g` are `public` functions in `C`)
- `condition ? C.f : C.g` (where `f` and `g` are both `internal`)
- `condition ? C.f : L.f` / `condition ? L.f : C.f` (where `C.f` and `L.f` are both `internal`)

Reply:
I added the following tests: 
- `condition ? C.f : C.f`
- `condition ? C.f : C.g` (where `f` and `g` are both `external`)
- `condition ? C.f : D.f` (`external` functions from different contracts)
- `condition ? I.f : J.f` (functions from different interfaces)

First one, I added two versions, one in which the functions belong to the same contract and another one in which they belong to different contract. In the version where they are from the same contract, it produces `TypeError 9582`, which already happened in `0.8.20`. That is the case also for the tests which were suggested as semantic tests:
- `condition ? g : C.f` / `condition ? C.f : g` (`f` and `g` are `public` functions in `C`)
- `condition ? C.f : C.g` (where `f` and `g` are both `internal`)
- `condition ? C.f : L.f` / `condition ? L.f : C.f` (where `C.f` and `L.f` are both `internal`)

Because of that, I added them as syntax tests instead.


These remaining suggested tests are instances of the ICE @cameel mentioned [here](https://github.com/ethereum/solidity/pull/14367#discussion_r1262990597): 
- `condition ? L.f : L.g`
- `condition ? L.f : K.f` (functions from different libraries)

Besides those, there is still the problem discussed in this [comment](https://github.com/ethereum/solidity/pull/14367#discussion_r1261431279).

Reply:
You were getting `TypeError 9582` because you were trying to access `.selector` and internal functions don't have that. These cases would still work as semantic tests if you did something else with them, e.g. try to call them. We should convert these tests to that, since testing `TypeError 9582` here is pointless.

Though, given https://github.com/ethereum/solidity/pull/14367#discussion_r1265174602, it turns out that most of these (probably all cases of `TypeError 9582`) are not of `Declaration` kind so not that important. We mostly care about those that are.